spec/controllers/profile_controller_spec.rb

```diff
@@ -22,7 +24,66 @@
         expect(response.status).to eq(200)
       end
     end
-    
+
+    context 'changing passwords' do
+      it 'errors when new passwords dont match' do
+        @request.env["devise.mapping"] = Devise.mappings[:user]
+        @user = FactoryBot.create(:user_manager_user)
+        allow(request.env['warden']).to receive(:authenticate!).and_return(@user)
+        allow(controller).to receive(:current_user).and_return(@user)
+        sign_in @user
+        post :change_password, params: { password_change: { 'new_password1': 'wrong', 'new_password2': 'right' } }
+        expect(subject).to redirect_to(profile_path)
+        expect(flash.now[:error]).not_to be_nil
```

Comment:
```suggestion
```
no need to be checking for not nil, if the next expectation checks for a specific value ;)